script.js

```diff
@@ -1,4 +1,294 @@
+/* Beat That - Base
+Requirements
+1. There are 2 players and players take turns.
+2. When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+3. The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player        
+   specifies dice order.
+4. After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+Objectives:
+1. 1v1 player vs player dice roll game &&
+2. Roll 2 dice and concentate the 2 dice number together &&
+3. Players can choose order of dice to concentate (dice 1 + dice 2 || dice 2 + dice 1)
+
+What is required?
+1. Dice Roll helper function
+2. Global states for different gamemode
+3. Game function
+4. Instructions for user
+*/
+
+//GLOBAL VARIABLES
+var currentGameMode = "welcomeToBeatThat"; //Starting Game Mode
+var player1 = ""; //Player 1 IGN
+var player2 = ""; //Player 2 IGN
+
+var userChosenDiceOrder = ""; //User desired dice order, either 1 or 2 (input = userDiceOrder)
+
+// player 1 dice results, store in array for later use, prevent another diceroll to reset results too.
+var player1dice1Arr = [];
+var player1dice2Arr = [];
+// player 2 dice results, store in array for later use, prevent another diceroll to reset results too.
+var player2dice1Arr = [];
+var player2dice2Arr = [];
+
+var player1Score = ""; //Player 1 score at the end of turn
+var player2Score = ""; //Player 2 score at the end of turn
+
+// Dice Roll helper function
+var rollDice = function () {
+  // produces a float between 0 and 6
+  var randomFloat = Math.random() * 6;
+  // take off the decimal
+  var resultInteger = Math.floor(randomFloat) + 1;
+  console.log(`rollDice function, ${resultInteger}`);
+  return resultInteger;
+};
+
+// Player 1 Result Function
+var player1Result = function (player1, player2, userChosenDiceOrder) {
```

Comment:
the logic in your `player1Results` and `player2Results` function is very repetitive. think of how you can combine them into one function where in the function, you can detect whose turn it is and by extension, which array you should manipulate or get values from. this will shorten and organise your code greatly

Reply:
the reason why i emphasise the point of repetition so much is because in programming, we try not to repeat ourselves if possible - the reason for this is 1. for readability purpose, and 2. for code maintainability purpose, imagine you notice a bug here and need to change something, then you need to change it twice for the code sections you repeated. And this gets more complex if there are more instances of unnecessary repetition in your code.